### PR TITLE
[stdlib] `Int` comparison operator unit tests

### PR DESCRIPTION
**Pull request by MatveyF**
Added unit tests for `__lt__`, `__le__`, `__eq__`, `__ne__`, `__gt__`, and `__ge__` dunders of `Int`.

Please double check the tests in case I missed any corner cases. Other unit tests did not test interactions with other types so I did not do that either. 

Related to #3145

### CODE REVIEW

File: stdlib/test/builtin/test_int.mojo [hunk 1]
```diff
@@ -195,6 +195,54 @@ def test_int_uint():
...
+
+
+def test_eq():
+    assert_true(Int(5).__eq__(Int(5)))
+    assert_true(Int(0).__eq__(Int(0)))
+    assert_false(Int(0).__eq__(Int(1)))
+    assert_false(Int(5).__eq__(Int(10)))
+
+    var a: Int = 5
+    assert_true(a.__eq__(a))
```
Comment (laszlokindrat) [inline stdlib/test/builtin/test_int.mojo:219]: This case (and the one below for `__ne__`) is not really needed, could you please drop them?

File: stdlib/test/builtin/test_int.mojo [hunk 2]
```diff
@@ -195,6 +195,54 @@ def test_int_uint():
     assert_equal(0, int(u2))
 
 
+def test_lt():
```
Comment (laszlokindrat) [inline stdlib/test/builtin/test_int.mojo:198]: Instead of separate functions, could you please just put these into a single `test_comparisons`? We don't have a consistent strategy for testing these, so for now I would like to reduce the number of test functions.

**Review by laszlokindrat — APPROVED** [review]
Thanks for the patch! I have few nits, but LGTM otherwise. Could you please rebase on the latest nightly?

**Comment by MatveyF** [comment]
Hi @laszlokindrat! Thank you for your review, I address your suggestions and rebase the branch later today

**Comment by laszlokindrat** [comment]
!sync
